### PR TITLE
Builtin predciate implementation and unit tests

### PR DESCRIPTION
**Pull request by asimarslan**

### CODE REVIEW

**Comment by hasancelik** [comment]
verify

**Comment by cangencer** [comment]
I think it will be useful to have ToString() implementation for all the Predicates. Also is PR builder still not working?

File: Hazelcast.Net/Hazelcast.Core/Predicate.cs [hunk 1]
```diff
@@ -0,0 +1,864 @@
...
+        public static TruePredicate True()
+        {
+            return new TruePredicate();
+        }
+    }
+
+    /// <summary>
+    ///     SQL Predicate
+    /// </summary>
+    public class SqlPredicate : IPredicate<object, object>
```
Comment (cangencer) [inline Hazelcast.Net/Hazelcast.Core/Predicate.cs]: can we remove the generic types from IPredicate? There is no need for it.

File: Hazelcast.Test/Hazelcast.Client.Test/PredicatesTest.cs [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+        public virtual void TestAnd2()
+        {
+            var predicate = CreateAndPredicate();
+            map.Put("key", predicate);
+            map.Put("key", predicate);
+            Assert.AreEqual(predicate, map.Get("key"));
+        }
+
+        [Test]
+        public virtual void TestBetween2()
```
Comment (cangencer) [inline Hazelcast.Test/Hazelcast.Client.Test/PredicatesTest.cs]: could use more descriptive name for test here..  like TestBetween_Deserialization or something similar

**Comment by hasancelik** [comment]
verify

**Comment by cangencer** [comment]
👍

**Pull request merged**